finality-aleph/src/base_protocol/handler.rs

```diff
@@ -45,18 +102,98 @@ impl Handler {
             })
             .collect()
     }
-}
 
-#[cfg(test)]
-mod tests {
-    use crate::{aleph_primitives::Block, base_protocol::handler::Handler};
-
-    #[test]
-    fn initially_no_peers() {
-        let handler = Handler::new();
-        assert!(
-            handler.peers_info::<Block>().is_empty(),
-            "there should be no peers initially"
-        );
+    fn verify_connection(
+        &self,
+        peer_id: PeerId,
+        handshake: Vec<u8>,
+        is_inbound: bool,
+    ) -> Result<Roles, ConnectError> {
+        let handshake = BlockAnnouncesHandshake::<B>::decode_all(&mut &handshake[..])
+            .map_err(|_| ConnectError::BadlyEncodedHandshake)?;
```

Comment:
Having the underlying `codec` error might be informative, so please wrap it.

Reply:
done